### PR TITLE
K8s: update hack codegen script

### PR DESCRIPTION
**Pull request by charandas**
**What is this feature?**

Updates the hack script to the latest with the change to using our own fork for openapi go code generation.

**Why do we need this feature?**

We use a different abstraction called `Builder` in Grafana API Server package here where the openapi definitions aren't a monolith. Instead, each of the API Groups let their builder type expose only their own openapi definitions. So far, the codegen process was manual.

This changeset makes it less painful albeit at the cost of maintaining our own fork of the helper in `k8s.io/code-generator`. IMO, this is a price we should be able to pay. Maintaining this helper doesn't seem to be a steep task as it is only a thin wrapper around the underlying lib.

**Who is this feature for?**

Grafana App Platform

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/apis/common/v0alpha1/zz_generated.openapi.go [hunk 1]
```diff
@@ -19,58 +18,6 @@ func GetOpenAPIDefinitions(ref common.ReferenceCallback) map[string]common.OpenA
 	return map[string]common.OpenAPIDefinition{
 		"github.com/grafana/grafana/pkg/apis/common/v0alpha1.ObjectReference": schema_pkg_apis_common_v0alpha1_ObjectReference(ref),
 		"github.com/grafana/grafana/pkg/apis/common/v0alpha1.Unstructured":    Unstructured{}.OpenAPIDefinition(),
-		"k8s.io/apimachinery/pkg/apis/meta/v1.APIGroup":                       schema_pkg_apis_meta_v1_APIGroup(ref),
```
Comment (charandas) [inline pkg/apis/common/v0alpha1/zz_generated.openapi.go:22]: We cover these standard openapi defs in this helper [here](https://github.com/grafana/grafana/blob/main/pkg/services/grafana-apiserver/openapi.go#L29), so deleting their repetition in each of the group versions defs should not impact the actual outcome.

File: pkg/apis/common/v0alpha1/openapi_violation_exceptions.list
Comment (ryantxu) [inline pkg/apis/common/v0alpha1/openapi_violation_exceptions.list:1]: can we rename these to:
`zz_generated.openapi_violation_exceptions.list`

and if easy, delete it if empty?

File: hack/README.md [hunk 2]
```diff
@@ -14,14 +14,15 @@ total 0
 lrwxr-xr-x  1 ryan  staff  37 Oct  5 09:34 grafana -> /Users/ryan/workspace/grafana/grafana
 ```
 
-You can clone k8s [code-generator](https://github.com/kubernetes/code-generator) here and use `CODEGEN_PKG=<CODE-GENERATOR-GIT-ROOT>` when running the `update-codegen.sh` script.
+The current workflow is to run the following:
 
-The current workflow (sorry!) is to:
+```shell
```
Comment (ryantxu) [inline hack/README.md:19]:
```suggestion
```shell
go mod download

```

**Review by ryantxu — APPROVED** [review]
Looks great -- I expect we will continue to tweak it for a while, but this is a huge improvement.

File: hack/externalTools.go [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package hack
+
+import (
+	"k8s.io/code-generator/pkg/util"
+)
+
+var _ = util.CurrentPackage
```
Comment (charandas) [inline hack/externalTools.go:7]: @ryantxu Are you ok with this workaround to ensure `go mod tidy` doesn't try to delete `k8s.io/code-generator` from go.mod?

**Pull request merged**